bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/Auditor.java

```diff
@@ -343,167 +154,43 @@ public Auditor(final String bookieIdentifier,
-
-        numUnderreplicatedLedgersElapsedRecoveryGracePeriod = new Gauge<Integer>() {
-            @Override
-            public Integer getDefaultValue() {
-                return 0;
-            }
-
-            @Override
-            public Integer getSample() {
-                return numOfURLedgersElapsedRecoveryGracePeriodGuageValue.get();
-            }
-        };
-        this.statsLogger.registerGauge(ReplicationStats.NUM_UNDERREPLICATED_LEDGERS_ELAPSED_RECOVERY_GRACE_PERIOD,
-                numUnderreplicatedLedgersElapsedRecoveryGracePeriod);
-
-        numLedgersHavingNoReplicaOfAnEntry = new Gauge<Integer>() {
-            @Override
-            public Integer getDefaultValue() {
-                return 0;
-            }
-
-            @Override
-            public Integer getSample() {
-                return numLedgersHavingNoReplicaOfAnEntryGuageValue.get();
-            }
-        };
-        this.statsLogger.registerGauge(ReplicationStats.NUM_LEDGERS_HAVING_NO_REPLICA_OF_AN_ENTRY,
-                numLedgersHavingNoReplicaOfAnEntry);
-        numLedgersHavingLessThanAQReplicasOfAnEntry = new Gauge<Integer>() {
-            @Override
-            public Integer getDefaultValue() {
-                return 0;
-            }
-
-            @Override
-            public Integer getSample() {
-                return numLedgersHavingLessThanAQReplicasOfAnEntryGuageValue.get();
-            }
-        };
-        this.statsLogger.registerGauge(ReplicationStats.NUM_LEDGERS_HAVING_LESS_THAN_AQ_REPLICAS_OF_AN_ENTRY,
-                numLedgersHavingLessThanAQReplicasOfAnEntry);
-        numLedgersHavingLessThanWQReplicasOfAnEntry = new Gauge<Integer>() {
-            @Override
-            public Integer getDefaultValue() {
-                return 0;
-            }
-
-            @Override
-            public Integer getSample() {
-                return numLedgersHavingLessThanWQReplicasOfAnEntryGuageValue.get();
-            }
-        };
-        this.statsLogger.registerGauge(ReplicationStats.NUM_LEDGERS_HAVING_LESS_THAN_WQ_REPLICAS_OF_AN_ENTRY,
-                numLedgersHavingLessThanWQReplicasOfAnEntry);
-        numUnderReplicatedLedgers = new Gauge<Integer>() {
-            @Override
-            public Integer getDefaultValue() {
-                return 0;
-            }
-
-            @Override
-            public Integer getSample() {
-                return underReplicatedLedgersGuageValue.get();
-            }
-        };
-        this.statsLogger.registerGauge(NUM_UNDER_REPLICATED_LEDGERS_GUAGE, numUnderReplicatedLedgers);
+        this.auditorStats = new AuditorStats(statsLogger);
 
         this.bkc = bkc;
         this.ownBkc = ownBkc;
         this.admin = admin;
         this.ownAdmin = ownAdmin;
         initialize(conf, bkc);
 
-        executor = Executors.newSingleThreadScheduledExecutor(new ThreadFactory() {
+        ledgerCheckerExecutor = Executors.newSingleThreadExecutor(new ThreadFactory() {
```

Comment:
We can define and initialize it in AuditorCheckAllLedgersTask. And add a abstract method `shutdown()` in the AuditorTask, if we need to do some resource clean, we can override the shutdown(). In Auditor shutdown(), to invoke task.shutdown();

Reply:
addressed done.